### DISCUSSION TITLE
`rest` package now found in `msw` when building with NextJS 14

### DISCUSSION THREAD

**Discussion by devraj** [orval-labs/orval#1021] [Q&A] [answered]
I have been migrating my application to NextJS 14 and if I use `msw` and build the application the compiler complains that it can't find the `rest` package in `msw`.

```tsx
./api/admin/admin.msw.ts:14:10
Type error: Module '"msw"' has no exported member 'rest'.

  12 |  */
  13 | import { faker } from "@faker-js/faker";
> 14 | import { rest } from "msw";
     |          ^
  15 |
  16 | export const getGetSchoolYearsMock = () =>
  17 |   Array.from(
```

I have `msw` installed as a `devDependency`:

```
  "devDependencies": {
    "@faker-js/faker": "^8.2.0",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "autoprefixer": "^10",
    "eslint": "^8",
    "eslint-config-next": "14.0.0",
    "msw": "^2.0.3",
    "orval": "^6.19.1",
    "postcss": "^8",
    "prettier": "^3.0.3",
    "tailwindcss": "^3",
    "typescript": "^5"
  },
```

Wonder if this is something to do with `orval` or if someone could point me to the right direction, that would be amazing!

Thank you for your time.

---
Node version: v21.1.0
OS: macOS 14
Framework: NextJS 14.0.1
orval: 6.19.1

**Comment by melloware** [comment]
@Will-Mann-16 just added MSW 2.0 support to the next version he may be able to answer

**Reply by Will-Mann-16** [reply, marked as answer]
Yeah this is because you're on MSW 2.0 where they removed the `rest` operator in favour of `http`. I've upgraded main to reflect this, just waiting on a release. In the meantime if you downgrade msw to v1 it'll work just fine.
